Neos.Media/Classes/Domain/Service/ThumbnailService.php

```diff
@@ -92,11 +92,6 @@ public function getThumbnail(AssetInterface $asset, ThumbnailConfiguration $conf
             if ($asset->getWidth() === null && $asset->getHeight() === null) {
                 return $asset;
             }
-            $maximumWidth = ($configuration->getMaximumWidth() > $asset->getWidth()) ? $asset->getWidth() : $configuration->getMaximumWidth();
-            $maximumHeight = ($configuration->getMaximumHeight() > $asset->getHeight()) ? $asset->getHeight() : $configuration->getMaximumHeight();
-            if ($configuration->isUpScalingAllowed() === false && $maximumWidth === $asset->getWidth() && $maximumHeight === $asset->getHeight()) {
-                return $asset;
-            }
```

Comment:
@aertmann I first made it configurable but then saw that a check happens anyway at a later time. The code gets quite bulky when we add additional option and I see it actually as a bug if Neos decides here that images don't need any processing at all only based on the image size.